bcgov/nr-compliance-enforcement-cm
# chore: CE-556 HWCR drug administration information updates

Addresses [CE-556](https://env-ceds.atlassian.net/browse/CE-556). Changes to some fields and available options when entering Drug information for animals. 

**Note**: This PR is paired with an [equivalent PR in the non-CM repo](https://github.com/bcgov/nr-compliance-enforcement-cm/pull/101).

## Changes 🚧

- Added _Oral/nasal_ to options in 'Injection method'.
- Replaced an option in 'Fate of remaining...' with _Returned to vet_. 
- Removed _Adverse reactions_, _Discard method_, and _Amount discarded_ fields.
- Added _Additional comments_ text field.
- (In CM DB) Migrated removed fields' data into _Additional comments_ data. 

## How Has This Been Tested? 🔬

- [x] Manual testing, matching  from issue report.
- [x] Dummy data in DB for migration.

### To test 🔬

0. Ensure both this and `...cm` repo are on this issue's branch.
1. Run a local instance of the application, log in. 
2. Navigate to a HWC complaint and **Add animal**, then **Add drug**.
   - You should no longer see the removed fields. You should now see an 'Additional comments' open text field. 
   - Under 'Injection Method', you should now see a fourth option _Oral/nasal_
   - Under 'Fate of remaining drug in vial', you should not see _Reverse distribution_ but instead see _Returned to vet_
3. Fill out the fields, and save the 'Animal outcome'. 

### Checklist 📃


[CE-556]: https://env-ceds.atlassian.net/browse/CE-556?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-compliance-enforcement-cm-101-backend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement-cm/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement-cm/actions/workflows/merge.yml)
